bootstrap/src/basis/file.ml

```diff
@@ -201,38 +198,41 @@ let read_into_hlt buffer t =
   let _ = Read.(submit ~buffer t |> complete_hlt) in
   ()
 
-(* external write_inner: _?bytes -> t >os-> sint *)
-external write_inner: Stdlib.Bytes.t -> t -> sint = "hm_basis_file_write_inner"
-
-let rec write buffer t =
```

Comment:
We recursively called write until the entire buffer was written here. We don't handle the same problem in the new version. Make sure there isn't a subtle bug here.

Reply:
This was an actual problem. Fixed now.